manifests/dellhw_exporter.pp

```diff
@@ -0,0 +1,122 @@
+# @summary This module manages prometheus node dellhw_exporter
+# @param arch
+#  Architecture (amd64 or i386)
+# @param bin_dir
+#  Directory where binaries are located
+# @param download_extension
+#  Extension for the release binary archive
+# @param download_url
+#  Complete URL corresponding to the where the release binary archive can be downloaded
+# @param download_url_base
+#  Base URL for the binary archive
+# @param extra_groups
+#  Extra groups to add the binary user to
+# @param extra_options
+#  Extra options added to the startup command
+# @param group
+#  Group under which the binary is running
+# @param init_style
+#  Service startup scripts style (e.g. rc, upstart or systemd)
+# @param install_method
+#  Installation method: url or package (only url is supported currently)
+# @param manage_group
+#  Whether to create a group for or rely on external code for that
+# @param manage_service
+#  Should puppet manage the service? (default true)
+# @param manage_user
+#  Whether to create user or rely on external code for that
+# @param os
+#  Operating system (linux is the only one supported)
+# @param package_ensure
+#  If package, then use this for package ensure default 'latest'
+# @param package_name
+#  The binary package name - not available yet
+# @param purge_config_dir
+#  Purge config files no longer generated by Puppet
+# @param restart_on_change
+#  Should puppet restart the service on configuration change? (default true)
+# @param service_enable
+#  Whether to enable the service from puppet (default true)
+# @param service_ensure
+#  State ensured for the service (default 'running')
+# @param service_name
+#  Name of the dellhw exporter service (default 'dellhw_exporter')
+# @param user
+#  User which runs the service
+# @param version
+#  The binary release version
+# @param omreport_path
+#  The file path to the omReport executable (default "/opt/dell/srvadmin/bin/omreport")
+class prometheus::dellhw_exporter (
+  String $download_extension              = 'tar.gz',
```

Comment:
for all the string datatypes where an empty string isnt allowed, can you please update it to String[1]?

Reply:
sure, should I update the other modules? The lines you're referring to are a copy of lines from https://github.com/voxpupuli/puppet-prometheus/pull/463/files

Reply:
would be awesome if you like to do that!, but pleaee provide that change in another PR

Reply:
oh I missed it in my own PR, ups :(

Reply:
Does this look better now?

Reply:
yes, thanks!